File: jobfinder-server/src/main/java/ru/mityushin/jobfinder/server/model/Publication.java
package ru.mityushin.jobfinder.server.model;

import lombok.*;

import javax.persistence.*;
import java.util.UUID;

@Data @Builder
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Entity
@Table(name = "PUBLICATION")
public class Publication {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID", nullable = false)
    private Long id;

    @Column(name = "UUID", nullable = false, unique = true)
    private UUID uuid;

    @Column(name = "TITLE", length = 50, nullable = false)
    private String title;

    @Column(name = "DESCRIPTION", length = 50)
    private String description;

    @Column(name = "CONTENT", length = 255, nullable = false)
    private String content;

    @Column(name = "VISIBLE", nullable = false)
    private Boolean visible;

    @Column(name = "DELETED", nullable = false)
    private Boolean deleted;

}

File: jobfinder-client/src/app/_models/publication.ts
export class Publication {
  uuid: string;
  authorUuid: string;
  title: string;
  description: string;
  content: string;
  visible: boolean;

  constructor(uuid: string,
              authorUuid: string,
              title: string,
              description: string,
              content: string,
              visible: boolean) {
    this.uuid = uuid;
    this.authorUuid = authorUuid;
    this.title = title;
    this.description = description;
    this.content = content;
    this.visible = visible;
  }
}
